pkg/providers/k8s/namespace/namespace_provider.go

```diff
@@ -103,7 +103,21 @@ func (c *watchingProvider) Init(ctx context.Context) error {
 }
 
 func (c *watchingProvider) initWatchingNamespacesByLabels(ctx context.Context) error {
-	labelSelector := metav1.LabelSelector{MatchLabels: c.watchingLabels}
+	valuesByKey := make(map[string][]string)
+	for key, value := range c.watchingLabels {
+		valuesByKey[key] = append(valuesByKey[key], value)
+	}
+	var matchExpressions []metav1.LabelSelectorRequirement
+	for key, values := range valuesByKey {
+		requirement := metav1.LabelSelectorRequirement{
+			Key:      key,
+			Operator: metav1.LabelSelectorOpIn,
+			Values:   values,
+		}
+		matchExpressions = append(matchExpressions, requirement)
+	}
+
+	labelSelector := metav1.LabelSelector{MatchExpressions: matchExpressions}
```

Comment:
I opted for `MatchLabels` over `MatchExpressions` in the LabelSelector criteria. This choice enables us to select multiple label values and facilitates support for a `namespace_selector` with multiple values for the same label: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements